New warning on gcc16

From the CRAN warnings:

```
g++-16 -std=gnu++17 -I"/home/hornik/tmp/R.check/r-devel-gcc/Work/build/include" -DNDEBUG -I../src/c -I../src/c/include -I../src/c/vendor/ -I../src/c/vendor/portable-snippets/include/ -I../src/c/vendor/fmt/include -I/usr/include/postgresql  -DADBC_EXPORT="" -DFMT_HEADER_ONLY=1  -I/usr/local/include -D_FORTIFY_SOURCE=3   -fpic  -g -O2 -Wall -pedantic -mtune=native   -c c/driver/postgresql/statement.cc -o c/driver/postgresql/statement.o
gcc-16 -I"/home/hornik/tmp/R.check/r-devel-gcc/Work/build/include" -DNDEBUG -I../src/c -I../src/c/include -I../src/c/vendor/ -I../src/c/vendor/portable-snippets/include/ -I../src/c/vendor/fmt/include -I/usr/include/postgresql  -DADBC_EXPORT="" -DFMT_HEADER_ONLY=1  -I/usr/local/include -D_FORTIFY_SOURCE=3   -fpic  -g -O2 -Wall -Wstrict-prototypes -pedantic -mtune=native  -c c/vendor/nanoarrow/nanoarrow.c -o c/vendor/nanoarrow/nanoarrow.o
In file included from /usr/include/stdio.h:970,
                 from c/vendor/nanoarrow/nanoarrow.c:21:
In function ‘snprintf’,
    inlined from ‘PrivateArrowSchemaToString’ at c/vendor/nanoarrow/nanoarrow.c:2056:20:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:68:10: warning: null destination pointer [-Wformat-truncation=]
   68 |   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   69 |                                    __glibc_objsize (__s), __fmt,
      |                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   70 |                                    __va_arg_pack ());
      |                                    ~~~~~~~~~~~~~~~~~
```

I think the issue is that `ArrowSchemaToString()` can return `nullptr`, and is sometimes used in a printf string. I think this functionally doesn't happen because the `ArrowType` in question is provided by an `ArrowSchemaView`, whose initializer never assigns an unknown type; however, we should still fix this.

--- a/python/src/nanoarrow/_array.pyx
+++ b/python/src/nanoarrow/_array.pyx
@@ -143,8 +143,7 @@ cdef class CArrayView:
     @property
     def storage_type(self):
         cdef const char* type_str = ArrowTypeString(self._ptr.storage_type)
-        if type_str != NULL:
-            return type_str.decode('UTF-8')
+        return type_str.decode('UTF-8')
 
     @property
     def layout(self):

--- a/python/src/nanoarrow/_schema.pyx
+++ b/python/src/nanoarrow/_schema.pyx
@@ -594,18 +594,12 @@ cdef class CSchemaView:
     @property
     def type(self) -> str:
         cdef const char* type_str = ArrowTypeString(self._schema_view.type)
-        if type_str != NULL:
-            return type_str.decode()
-        else:
-            raise ValueError("ArrowTypeString() returned NULL")
+        return type_str.decode()
 
     @property
     def storage_type(self) -> str:
         cdef const char* type_str = ArrowTypeString(self._schema_view.storage_type)
-        if type_str != NULL:
-            return type_str.decode()
-        else:
-            raise ValueError("ArrowTypeString() returned NULL")
+        return type_str.decode()
 
     @property
     def dictionary_ordered(self) -> Union[bool, None]:

--- a/src/nanoarrow/common/inline_types.h
+++ b/src/nanoarrow/common/inline_types.h
@@ -473,7 +473,7 @@ enum ArrowType {
 /// \brief Get a string value of an enum ArrowType value
 /// \ingroup nanoarrow-utils
 ///
-/// Returns NULL for invalid values for type
+/// Returns "<unknown type identifier>" for invalid values for type
 static inline const char* ArrowTypeString(enum ArrowType type);
 
 static inline const char* ArrowTypeString(enum ArrowType type) {
@@ -569,7 +569,7 @@ static inline const char* ArrowTypeString(enum ArrowType type) {
     case NANOARROW_TYPE_LARGE_LIST_VIEW:
       return "large_list_view";
     default:
-      return NULL;
+      return "<unknown type identifier>";
   }
 }
 

